### PR TITLE
Introduce analytics id

### PR DESCRIPTION
**Pull request by olivereisenhut**

### CODE REVIEW

File: BMM.Core/Implementations/Analytics/AppCenterAnalytics.cs [hunk 1]
```diff
@@ -24,8 +25,8 @@ public void LogEvent(string eventName)
         public void LogEvent(string eventName, IDictionary<string, object> parameters)
         {
             var user = _userStorage.GetUser();
-            if (!parameters.ContainsKey("PersonId") && user != null)
-                parameters.Add("PersonId", _userStorage.GetUser().PersonId);
+            if (!parameters.ContainsKey(nameof(User.AnalyticsIdentifier)) && user != null)
```
Comment (kkuepper) [inline BMM.Core/Implementations/Analytics/AppCenterAnalytics.cs:28]: Not sure if we should change that right now

File: BMM.Core/Implementations/Security/ClaimUserInformationExtractor.cs [hunk 2]
```diff
@@ -12,30 +15,52 @@ public class ClaimUserInformationExtractor : IClaimUserInformationExtractor
 
         public User ExtractUser(ClaimsPrincipal claims)
```
Comment (kkuepper) [inline BMM.Core/Implementations/Security/ClaimUserInformationExtractor.cs:16]: do we actually still need this?

File: BMM.Core/Translation/en/main.json [hunk 3]
```diff
@@ -280,6 +280,7 @@
     "OptionAppVersionHeader": "App version",
     "AppInfoTitle": "App information",
     "AppInfoAppVersion": "App version: ",
+    "AppInfoManufacturer": "Manufacturer: ",
```
Comment (kkuepper) [inline BMM.Core/Translation/en/main.json:283]: Where is this being used?

Comment (olivereisenhut) [reply BMM.Core/Translation/en/main.json:283]: Profile Screen -> App Version. The translation was never added.

File: BMM.Core/Implementations/Security/ClaimUserInformationExtractor.cs (hunk 2 above)
Comment (olivereisenhut) [reply BMM.Core/Implementations/Security/ClaimUserInformationExtractor.cs:16]: I merged it now

File: BMM.Core/Implementations/Feedback/EmailContacter.cs [hunk 4]
```diff
@@ -32,7 +32,7 @@ public async Task Contact()
             sb.AppendLine("Device version: " + _deviceInfo.VersionString);
             sb.AppendLine("Full name: " + user.FullName);
             sb.AppendLine("Username: " + user.Username);
-            sb.AppendLine("PersonId: " + user.PersonId);
+            sb.AppendLine("AnalyticsIdentifier: " + user.AnalyticsIdentifier);
```
Comment (kkuepper) [inline BMM.Core/Implementations/Feedback/EmailContacter.cs:35]: I'm not sure if we should always send it or if the user has to voluntarily reveal that information

**Comment by olivereisenhut** [comment]
Logging the age group is missing.

File: BMM.Core/Implementations/Analytics/AppCenterAnalytics.cs (hunk 1 above)
Comment (kkuepper) [reply BMM.Core/Implementations/Analytics/AppCenterAnalytics.cs]: introduced a remote setting to control that

File: BMM.Core/Implementations/Analytics/AppCenterAnalytics.cs [hunk 5]
```diff
@@ -24,8 +28,19 @@ public void LogEvent(string eventName)
...
+                    parameters.Add(nameof(User.AnalyticsId), user.AnalyticsId);
+            }
+            else
+            {
+                if (!parameters.ContainsKey(nameof(user.PersonId)) && user != null)
+                    parameters.Add(nameof(user.PersonId), user.PersonId);
+            }
+
+            if (user?.Age != null)
+                parameters.Add(nameof(user.Age), user.Age);
```
Comment (akrol95) [inline BMM.Core/Implementations/Analytics/AppCenterAnalytics.cs:43]: I would do it that way, to not copy the same check:

```
        public void LogEvent(string eventName, IDictionary<string, object> parameters)
        {
            var user = _userStorage.GetUser();

            void AddParameter(string key, object value)
            {
                if (!parameters.ContainsKey(key) && user != null)
                    parameters.Add(key, value);
            }

            if (_remoteConfig.UseAnalyticsId)
                AddParameter(nameof(User.AnalyticsId), user.AnalyticsId);
            else
                AddParameter(nameof(user.PersonId), user.PersonId);

            (...)
        }

```

File: BMM.Core/Implementations/PlayObserver/PlayStatistics.cs [hunk 6]
```diff
@@ -243,6 +243,10 @@ public async Task WriteEvent(TrackPlayedEvent ev)
                 {"sentAfterStartup", ev.SentAfterStartup},
                 {"playbackOrigin", ev.PlaybackOrigin}
             };
+            if (_config.UseAnalyticsId)
+                dict.Add(nameof(User.AnalyticsId), ev.AnalyticsId);
+            else
+                dict.Add(nameof(User.PersonId), ev.PersonId);
```
Comment (akrol95) [inline BMM.Core/Implementations/PlayObserver/PlayStatistics.cs:249]: Add one line space above and below this if..else statement.

**Review by akrol95 — APPROVED** [review]

File: BMM.Core/Implementations/Analytics/AppCenterAnalytics.cs (hunk 5 above)
Comment (kkuepper) [reply BMM.Core/Implementations/Analytics/AppCenterAnalytics.cs]: nice. refactored it

File: BMM.Core/Implementations/PlayObserver/PlayStatistics.cs (hunk 6 above)
Comment (kkuepper) [reply BMM.Core/Implementations/PlayObserver/PlayStatistics.cs:250]: done

**Pull request merged**